# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Opcodes.h
# Commit: d9c1e44eb0ad
# Full Hash: d9c1e44eb0addfc3e44ae299c86381128aa08e7f
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-15 21:49:48
# Regressor Bug: 1601599
# File Overlap Count: 1
# Description:
#   Bug 1601599 part 4 - Support Ion OSR at all loops. r=tcampbell
#   
#   Depends on D56700
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56701
# ==============================================================================

diff -r 93842c3971f8 -r d9c1e44eb0ad js/src/vm/Opcodes.h
--- a/js/src/vm/Opcodes.h	Sun Dec 15 11:33:36 2019 +0000
+++ b/js/src/vm/Opcodes.h	Sun Dec 15 11:34:34 2019 +0000
@@ -1112,17 +1112,14 @@
     /*
      * This opcode is the target of the backwards jump for some loop.
      *
-     * The uint8 argument is a bitfield. The lower 7 bits of the argument
-     * indicate the loop depth. This value starts at 1 and is just a hint:
-     * deeply nested loops all have the same value. The upper bit is set if Ion
-     * should be able to OSR at this point, which is true unless there is
-     * non-loop state on the stack.
+     * The depthHint value is a loop depth hint for Ion. It starts at 1 and
+     * deeply nested loops all have the same value.
      *
      * See JSOP_JUMPTARGET for the icIndex operand.
      *
      *   Category: Statements
      *   Type: Jumps
-     *   Operands: uint32_t icIndex, uint8_t BITFIELD
+     *   Operands: uint32_t icIndex, uint8_t depthHint
      *   Stack: =>
      */ \
     MACRO(JSOP_LOOPHEAD, 109, "loophead", NULL, 6, 0, 0, JOF_LOOPHEAD) \
